components/service/glean/src/main/java/mozilla/components/service/glean/storages/TimespansStorageEngine.kt

```diff
@@ -121,6 +121,30 @@ internal open class TimespansStorageEngineImplementation(
         }
     }
 
+    /**
+     * Set the elapsed time explicitly.
+     *
+     * @param metricData the metric information for the timespan
+     * @param timeUnit the time unit we want the data in when snapshotting
+     * @param elapsedNanos the time to record, in nanoseconds
+     */
+    @Synchronized
+    fun set(
+        metricData: CommonMetricData,
+        timeUnit: TimeUnit,
+        elapsedNanos: Long
+    ) {
+        // Look for the start time: if it's there, commit the timespan.
+        val timespanName = metricData.identifier
+
+        // Store the time unit: we'll need it when snapshotting.
+        timeUnitsMap[timespanName] = timeUnit
+
+        super.recordMetric(metricData, elapsedNanos, timeUnit) { currentValue, newValue ->
```

Comment:
I think something will complain about `currentValue` not being used. Please replace it with `_`.